cso-core/src/expression.rs

```diff
@@ -10,9 +11,7 @@ pub trait ScalarExpression: AsAny + Debug + Clone {
 
     fn equal(&self, other: &dyn ScalarExpression) -> bool;
 
-    fn derive_used_columns(&self) {
-        // todo!()
-    }
+    fn derive_used_columns(&self, col_set: &mut ColumnRefSet);
```

Comment:
Why pass in a mutable reference instead of returning a ColumnRefSet directly?

Reply:
to reduce memory allocation